rx-java/src/main/java/io/vertx/rx/java/ReadStreamSubscriber.java

```diff
@@ -0,0 +1,182 @@
+package io.vertx.rx.java;
+
+import io.vertx.core.Handler;
+import io.vertx.core.streams.ReadStream;
+import rx.Observable;
+import rx.Subscriber;
+
+import java.util.ArrayDeque;
+import java.util.function.Function;
+
+/**
+ * An RxJava {@code Subscriber} that turns an {@code Observable} into a {@link ReadStream}.
+ * <p>
+ * The stream implements the {@link #pause()} and {@link #resume()} operation by maintaining
+ * a buffer of {@link #BUFFER_SIZE} elements between the {@code Observable} and the {@code ReadStream}.
+ * <p>
+ * When the subscriber is created it requests {@code 0} elements to activate the subscriber's back-pressure.
+ * Setting the handler initially on the {@code ReadStream} triggers a request of {@link #BUFFER_SIZE} elements.
+ * When the item buffer is half empty, new elements are requested to fill the buffer back to {@link #BUFFER_SIZE}
+ * elements.
+ * <p>
+ * The {@link #endHandler(Handler<Void>)} is called when the {@code Observable} is completed or has failed and
+ * no pending elements, emitted before the completion or failure, are still in the buffer, i.e the handler
+ * is not called when the stream is paused.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class ReadStreamSubscriber<R, J> extends Subscriber<R> implements ReadStream<J> {
+
+  private static final Runnable NOOP_ACTION = () -> {};
+  private static final Throwable DONE_SENTINEL = new Throwable();
+
+  public static final int BUFFER_SIZE = 16;
+
+  public static <R, J> ReadStream<J> asReadStream(Observable<R> observable, Function<R, J> adapter) {
```

Comment:
And what would the adapter function be? Won't it be identity again? Do we want to convert the buffer in HttpRequest#send ? How does codegen know how to convert?